allennlp/data/dataset_readers/reading_comprehension/squad.py

```diff
@@ -100,17 +106,28 @@ def text_to_instance(self,  # type: ignore
                 logger.debug("Answer: %s", passage_text[char_span_start:char_span_end])
             token_spans.append((span_start, span_end))
 
-        return util.make_reading_comprehension_instance(self._tokenizer.tokenize(question_text),
-                                                        passage_tokens,
-                                                        self._token_indexers,
-                                                        passage_text,
-                                                        token_spans,
-                                                        answer_texts)
+        if self._multiparagraph:
+            return util.make_multi_paragraph_reading_comprehension_instance(
+                    self._tokenizer.tokenize(question_text),
+                    [passage_tokens],
```

Comment:
For squad we only pass in the paragraph which contains the answer? 
Does this not defy the point of this PR, as we are suppose to be training on a model which has paragraphs which do not contain the answer?

Reply:
the squad dataset only has one paragraph for each question, what else should I be passing in?

Reply:
Chris also tested on versions where you have to pick the paragraph from SQuAD; we haven't gotten that far yet.

Reply:
In squad dataset there is multiple paragraphs from a single document. for example there are a lot of superbowl questions which all derive from the single article: https://rajpurkar.github.io/SQuAD-explorer/explore/1.1/dev/Super_Bowl_50.html?model=r-net+%20(ensemble)%20(Microsoft%20Research%20Asia)

Perhaps we could pass all paragraphs from the section same section too as long as they do not contain the answer. 

Although for the moment I think what you have is good enough :)

Reply:
Yes, that's basically what Chris did, and I think we'll want to have that eventually.  It doesn't have to be in this PR, though.